src/components/Sidebars/Payback.js

```diff
@@ -31,67 +31,93 @@ const Payback = ({ cdpId, reset }) => {
   const [daiBalance, setDaiBalance] = useState(0);
   const [liquidationPrice, setLiquidationPrice] = useState(0);
   const [collateralizationRatio, setCollateralizationRatio] = useState(0);
-  const [allowanceState, setAllowanceState] = useState({
-    proxyAddress: '',
-    userStartedWithoutAllowance: false,
+
+  const [userState, setUserState] = useState({
+    proxyAddress: null,
     allowance: null,
-    loading: false
+    startedWithoutProxy: false,
+    startedWithoutAllowance: false,
+    allowanceLoading: false,
+    proxyLoading: false
   });
 
   useEffect(() => {
-    const checkProxy = async () => {
-      const proxy = await maker.service('proxy').ensureProxy();
-      return proxy;
-    };
-
     const init = async () => {
-      const proxy = await checkProxy();
-      const allowance = await checkAllowance({ address: proxy });
-      setAllowanceState({
-        ...allowanceState,
-        proxyAddress: proxy,
-        userStartedWithoutAllowance: !allowance,
-        allowance
-      });
+      const proxyAddress = await checkProxy();
+      if (proxyAddress) {
+        const allowance = await checkAllowance(proxyAddress);
+        setUserState({
+          ...userState,
```

Comment:
since you're always calling `setUserState` in a way that preserves existing state, you may as well factor out this use of the spread operator so you don't have to keep repeating it. i.e.:
```js
const [userState, updateUserState] = useReducer((state, updates) => ({ ...state, ...updates }), {});
// that second argument is initial state
// i left it empty because all of your initial values are falsy, 
// so there's no need to declare them

updateUserState({ proxyAddress, allowance, startedWithoutAllowance: !allowance });
updateUserState({ proxyLoading: true }); // etc.
```